modules/ListView.js

```diff
@@ -349,10 +367,31 @@
      * @method update
      */
     update: function() {
-      var newViews = this.__createChildViews(),
-        removedViews = this.__removeStaleChildViews();
-      if (!_.isEmpty(newViews) || !_.isEmpty(removedViews)) {
+      var view = this,
+        renderNeeded = false,
+        oldViews = this.getItemViews(),
+        newViews = this.__createItemViews(),
+        staleViews = this.__getStaleItemViews(),
+        sizeOfOldViews = _.size(oldViews),
+        sizeOfNewViews = _.size(newViews),
+        sizeOfStaleViews = _.size(staleViews),
+        changes = sizeOfNewViews + sizeOfStaleViews,
+        percentChange = changes / Math.max((sizeOfOldViews - sizeOfStaleViews + sizeOfNewViews), 1),
+        threshold = this.updateThreshold || 0.5;
+      if (!changes) {
+        this.reorder();
+        return;
+      }
+      // A switch from empty to not empty or vise versa, needs a new render
+      renderNeeded = (!sizeOfOldViews && sizeOfNewViews) || (sizeOfOldViews && sizeOfOldViews === sizeOfStaleViews && !sizeOfNewViews);
+      if (renderNeeded || percentChange >= threshold) {
```

Comment:
Readability: make this part of the separate lines leading up to renderNeeded.  This line should just be if (renderNeeded).

My rule of thumb is one action per line.